### ISSUE TITLE
subset error

### ISSUE THREAD

**Issue by DenizEsen** [satijalab/seurat#2973]
Hello,

I have been running into an "Error: Under current subsetting parameters, the default assay will be removed. Please adjust subsetting parameters or change default assay." while trying to subset my data. The code I use is simply `subset(object, gene>0)`, or `subset(object, metadata id == factor,` which work for another dataset I have previously analyzed. Aside from subsetting, I would like to use FindMarkers as `FindMarkers(object, ident.1 = "factor", group.by = "metadata id", subset.ident = "cluster id")`, and the addition of subset.ident parameter returns the same error. 

The same issue was raised in #1444, and I have tried reinstalling Seurat package and tried to run my code in 2 different computers. I have tried with several genes, which are expressed in at least a few thousand cells. It seems like my issue is with the Seurat object itself. 

WhichCells function does return me cell names when used as `WhichCells(object, expression = gene>0)`
My active assay is RNA and active ident is cluster names, as is in the other data I have used the same code with.

[The object is one I have downloaded from a public repository](https://datashare.is.ed.ac.uk/handle/10283/3433), originally v2.2.1. UpdateSeuratObject did not work either, giving me the "Error: None of the features provided are in this Assay object". I had resorted to creating a new Seurat object from its data slot, then replacing the count matrix in the new one with the original, adding the metadata too and proceeded with my analysis as usual post normalization stage. Was this a bad idea and could it be linked to the error I am getting from the subset function?

> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.4 LTS
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:
[1] ggplot2_3.3.0 Seurat_3.1.5

**Comment by DenizEsen** [comment]
I have identified the source of the error. It seems some feature names had problematic syntax. I removed them from the object as `object[c(1:a-1,a+1:b-1, b+1:n),]` etc., where a and b are indexes of incompatible gene names. subset function worked without any issues afterwards.

The offenders were: 
- RP11-442N24--B.1
- RP4-633O19--A.1
- RP11-544L8--B.4
- RP11-1157N2--B.2
- RP11-59D5--B.2 
- XXyac-YX65C7-A.2

I have considered renaming these genes to eliminate the problem, but saw there are no trivial methods of renaming features and it is better to rename them in the raw data, then perform the analysis. Since I'm mostly done after this point, I decided to lazily remove these genes as they probably will not affect my analysis anyways.
